OpenRA.Mods.Common/UtilityCommands/ExtractEmmyLuaAPI.cs

```diff
@@ -0,0 +1,342 @@
+			Console.WriteLine("-- See https://docs.openra.net/en/latest/release/lua/ for human readable documentation.");
+
+			Console.WriteLine();
+			Console.WriteLine();
+			Console.WriteLine("--");
+			Console.WriteLine("-- SECTION Manually added:");
+			Console.WriteLine("--");
+			Console.WriteLine();
+			Console.WriteLine();
+
+			WriteManual();
+
+			Console.WriteLine();
+			Console.WriteLine("--");
+			Console.WriteLine("-- SECTION Script API global methods:");
+			Console.WriteLine("--");
+			Console.WriteLine();
+			Console.WriteLine();
+
+			var globalTables = Game.ModData.ObjectCreator.GetTypesImplementing<ScriptGlobal>().OrderBy(t => t.Name);
+			WriteGlobals(globalTables);
+
+			Console.WriteLine();
+			Console.WriteLine("--");
+			Console.WriteLine("-- SECTION Script API object properties:");
+			Console.WriteLine("--");
+			Console.WriteLine();
+			Console.WriteLine();
+
+			var actorProperties = Game.ModData.ObjectCreator.GetTypesImplementing<ScriptActorProperties>();
+			WriteScriptProperties(typeof(Actor), actorProperties);
+
+			var playerProperties = Game.ModData.ObjectCreator.GetTypesImplementing<ScriptPlayerProperties>();
+			WriteScriptProperties(typeof(Player), playerProperties);
+		}
+
+		static void WriteManual()
+		{
+			Console.WriteLine("--- This function is triggered once, after the map is loaded.");
+			Console.WriteLine("function WorldLoaded() end");
+			Console.WriteLine();
+			Console.WriteLine("--- This function will hit every game tick which by default is every 40 ms.");
+			Console.WriteLine("function Tick() end");
+			Console.WriteLine();
+
+			Console.WriteLine("---A subset of all ActorInit implementations.");
+			Console.WriteLine("---@class initTable");
+			Console.WriteLine("---@field Location cpos");
+			Console.WriteLine("---@field Owner player");
+			Console.WriteLine("---@field Facing wangle");
+			Console.WriteLine("---@field CreationActivityDelay integer");
+			Console.WriteLine("---@field CenterPosition wpos");
+			Console.WriteLine("---@field Faction string");
+			Console.WriteLine("---@field EffectiveOwner player");
+			Console.WriteLine("---@field Stance userdata");
+			Console.WriteLine("---@field FreeActor boolean");
+			Console.WriteLine("---@field ParentActor actor");
+			Console.WriteLine("---@field LineBuildDirection userdata");
+			Console.WriteLine("---@field LineBuildParent string[]");
+			Console.WriteLine("---@field RuntimeCargo actor[]");
+			Console.WriteLine("---@field Cargo string[]");
+			Console.WriteLine("---@field DeployState userdata");
+			Console.WriteLine("---@field Experience integer");
+			Console.WriteLine("---@field Health integer");
+			Console.WriteLine("---@field HuskSpeed integer");
+			Console.WriteLine("---@field Plug string");
+			Console.WriteLine("---@field ProductionSpawnLocation cpos");
+			Console.WriteLine("---@field ScriptTags string[]");
+			Console.WriteLine("---@field TurretFacing wangle");
+			Console.WriteLine("---@field SpawnedByMap string");
```

Comment:
How were those generated? Did you take a subset by hand? Fwiw exposing/advertising `SpawnedByMap` to Lua does look very weird. (I also haven't tested if all these inits here work.)

Reply:
These are guaranteed to bitrot and get out of sync with reality. It should be possible to fetch these using reflection (which is what the actual Lua API does after all).

Reply:
Removed them.

Reply:
They are now being reflected.